src/views/validated-designs/server.ts

```diff
@@ -27,7 +33,7 @@ function loadMetadata(path: string): { title: string; description: string } {
 	}
 }
 
-export function getHvdLandingProps(): ValidatedDesignsLandingProps | null {
+export function getHvdCategoryGroups(): HvdCategoryGroup[] {
```

Comment:
What do you think about breaking this function in two? One function is the code in the `if` block and the other is the code in the `else if` block.

Reply:
I am a bit indifferent about breaking it up, but I don't think breaking it up would clean up the code or how this function is used at all. Because ultimately separating out the "unpure" parts, the parts that could fail, from the "pure" parts wouldn't do much because both functions would always have to be called together anyway. Here is why.

I designed this function to both load the data and format it, because I need that data in 3 different places for Static Site Generation and to support Incremental Static Regeneration. This function recreates the data each time because when the HVD Docs repo is updated we will be firing a GitHub action that marks paths in our HVD docs as stale and will redownload the repo and rebuild those paths. (As opposed to requiring a new dev-portal build each time they are updated.)

The 3 places are below. These are wrappers around this function modify it's data slightly to fit the different required data needed:
* [guides list props build](https://github.com/hashicorp/dev-portal/blob/7d23744ff4b67d70d0817a5c4ada23f480a8d885/src/pages/validated-designs/index.tsx#L17)
* [guide detail pages static paths build](https://github.com/hashicorp/dev-portal/blob/7d23744ff4b67d70d0817a5c4ada23f480a8d885/src/pages/validated-designs/%5B...slug%5D/index.tsx#L16)
* [guide detail static props build](https://github.com/hashicorp/dev-portal/blob/7d23744ff4b67d70d0817a5c4ada23f480a8d885/src/pages/validated-designs/%5B...slug%5D/index.tsx#L39)